index.bs

```diff
@@ -2302,6 +2349,93 @@ To <dfn>parse trigger specs</dfn> given a [=map=] |map|, a [=moment=]
 Issue: Invoke [=parse summary buckets=] and [=parse summary window operator=]
 from this algorithm.
 
+To <dfn>parse max event states</dfn> from a [=map=] |map|,
+a [=source type=] |sourceType|,
+a possibly null 32-bit unsigned integer |attributionScopeLimit|,
+and a [=randomized response output configuration=] |config|:
+1. If |attributionScopeLimit| is null:
+    1. If |map|["`max_event_states`"] [=map/exists=], return an error.
+    1. Return null.
+1. Let |states| be the [=obtain a set of possible trigger states|number of possible trigger states=] with |config|.
+1. If |sourceType| is "<code>[=source type/event=]</code>" and |states| is greater than [=default max event states=]
+    and |map|["`max_event_states`"] does not [=map/exist=], return an error.
+1. Let |maxEventStates| be the result of running
+    [=parse an optional 32-bit unsigned integer=] with |map|,
+    "`max_event_states`", and [=default max event states=].
+1. If |maxEventStates| is an error or |maxEventStates| is equal to 0 or |maxEventStates| is greater than [=max trigger-state cardinality=], return an error.
+1. If |sourceType| is "<code>[=source type/event=]</code>" and |states| is greater than |maxEventStates|, return an error.
+1. Return |maxEventStates|.
+
+To <dfn>parse attribution scopes</dfn> from a [=map=] |map| and a possibly null 32-bit unsigned integer |attributionScopeLimit|:
+1. If |map|["`attribution_scopes`"] does not [=map/exist=], return null.
```

Comment:
The method is shared with trigger attribution scopes as well. For trigger `null` is reserved to match all sources (with or without scopes).

if we return empty as default, it would cause all sources rejected.

Reply:
In that case, maybe we can return the empty list for sources but null for triggers?

Reply:
Done.